app.py

```diff
@@ -138,6 +138,20 @@ def register_proposal(caller_identity):
     if proposal is None:
         return jsonify(error='missing service_proposal'), 400
 
+    service_type = proposal.get('service_type', None)
+
+    # check if the service type is defined.
+    # If defined, check if it's in the possible enum values
+    if service_type is not None and \
+            service_type not in ServiceType.__members__:
```

Comment:
This service dont own `ServiceProposal` structures, so we decided not to validate here. API should be jsut dummy proxy for any structures.